torch/_decomp/decompositions.py

```diff
@@ -1263,3 +1263,100 @@ def kl_div_backward(
     if reduction == Reduction.MEAN.value:
         return grad_input / self.numel()
     return grad_input
+
+
+@register_decomposition(torch.ops.aten.upsample_bilinear2d.vec)
+def upsample_bilinear2d_vec(
+    input: Tensor,
+    output_size: Optional[List[int]],
+    align_corners: bool,
+    scale_factors: Optional[List[float]]
+) -> Tensor:
+    # get dimensions of original image
+    n_batch, n_channels, in_h, in_w = input.shape
+
+    if output_size is not None:
+        out_h = output_size[0]
+        out_w = output_size[1]
+    elif scale_factors is not None:
+        out_h = in_h * scale_factors[0]
+        out_w = in_w * scale_factors[1]
+
+    out_h_int = int(out_h)
+    out_w_int = int(out_w)
+
+    # Calculate horizontal and vertical scaling factor
+    if out_h > 1:
+        if align_corners:
+            h_scale_factor = (in_h - 1) / (out_h_int - 1)
+        else:
+            h_scale_factor = in_h / out_h
+    else:
+        h_scale_factor = 0
+
+    if out_w > 1:
+        if align_corners:
+            w_scale_factor = (in_w - 1) / (out_w_int - 1)
+        else:
+            w_scale_factor = in_w / out_w
+    else:
+        w_scale_factor = 0
+
+    b = torch.arange(n_batch, dtype=torch.int64)
+    c = torch.arange(n_channels, dtype=torch.int64)
+    i = torch.arange(out_h, dtype=torch.int64)
+    j = torch.arange(out_w, dtype=torch.int64)
+
+    if align_corners:
+        x = h_scale_factor * i
+        y = w_scale_factor * j
+    else:
+        x = h_scale_factor * (i + 0.5) - 0.5
+        x = torch.where(x > 0, x, 0)
+        y = w_scale_factor * (j + 0.5) - 0.5
+        y = torch.where(y > 0, y, 0)
+
+    x_floor = torch.floor(x)
+    x_ceil = torch.ceil(x)
+    x_ceil = torch.where(x_ceil > in_h - 1, in_h - 1, x_ceil)
+    y_floor = torch.floor(y)
+    y_ceil = torch.ceil(y)
+    y_ceil = torch.where(y_ceil > in_w - 1, in_w - 1, y_ceil)
+
+    result = torch.zeros(size=(n_batch, n_channels, out_h_int, out_w_int), dtype=input.dtype)
+    result_shape = result.shape
+
+    x_repeat = torch.repeat_interleave(x, out_w_int).view(out_h_int, out_w_int).repeat(n_batch, n_channels, 1, 1).view(result_shape)
```

Comment:
Why do we need all of these calls to repeat?  I'd expect to be able to handle this with broadcasting.

```
x_view = x.view(1, 1, len(x), 1)
y_view = x.view(1, 1, 1, len(y))
```